packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -104,12 +174,97 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
+	 *
+	 * 1. The event must match the requirements of the most specific relevant filter to it. This takes precedence over a more generic filter.
+	 * The less categories and longer the namespace within a filter, the more specific it is. Definining no categories is equivalant to defining all categories.
+	 *
+	 * 2. If a {@link TelemetryFilter} specifies both `categories` and a `namespace`, the event must match both.
+	 *
+	 * 3. If only `categories` or a `namespace` is provided, the event should match either one of them.
+	 *
+	 * 4. If a `namespace` pattern exception is specified in the {@link TelemetryFilter}, the event should not match the exception pattern.
+	 *
+	 * @param event - The telemetry event to check against the filters.
+	 *
+	 * @returns boolean `true` if the event matches any filter, otherwise `false`.
+	 */
 	private doesEventMatchFilter(event: ITelemetryBaseEvent): boolean {
 		for (const filter of this.config.filtering.filters ?? []) {
-			if (filter.category === event.category) {
+			if ("namespacePattern" in filter) {
+				if (event.eventName.startsWith(filter.namespacePattern)) {
+					// Found matching namespace pattern, since filters are ordered in most specific first,
+					// this is the most specific, relevant matching filter for the event.
+
+					// By default, if no categories are defined then any category is a valid match.
+					let doesFilterCategoriesMatch = true;
+					if ("categories" in filter && filter.categories.length > 0) {
+						doesFilterCategoriesMatch = false;
+						const matchingCategory = filter.categories.find(
+							(category) => category === event.category,
+						);
+						doesFilterCategoriesMatch = matchingCategory ? true : false;
+					}
+
+					if (doesFilterCategoriesMatch) {
+						// The most specific, relevant filter matches so no need to attempt to evaluate against other filters
+						// as long as the events namespace does not match any defined namespace exception.
+						if (filter.namespacePatternExceptions !== undefined) {
+							for (const patternException of filter.namespacePatternExceptions) {
+								if (event.eventName.startsWith(patternException)) {
+									return false;
+								}
+							}
+						}
+						return true;
+					} else {
+						return false;
+					}
+				}
+			}
+			// Filter only has categories defined
+			else if ("categories" in filter && filter.categories.length > 0) {
+				const doesFilterCategoriesMatch = filter.categories.find(
+					(category) => category === event.category,
+				);
+
+				// This filter specified no namespaces but it has a category match.
+				// Since filters are ordered by most specific first, we know that no previous
+				// filters with namespaces matched so we can return true.
+				if (doesFilterCategoriesMatch) {
+					return true;
+				} else {
+					continue;
+				}
+			} else {
 				return true;
 			}
 		}
+
 		return false;
 	}
+
+	public static sortTelemetryFilters(filters: TelemetryFilter[]): TelemetryFilter[] {
```

Comment:
This can be a non-exported free function in the module, I don't think it need to be exposed or part of the class.